### PR TITLE
imp: use storetypes instead of sdk types

### PR DESCRIPTION
**Pull request by faddat**
## Description

This PR changes the sdk.KVStore to storetypes.KVStore

Related PRs:

#3883 

### Commit Message / Changelog Entry

```bash
imp: use storetypes instead of sdk types
```

### CODE REVIEW

File: modules/core/module.go [hunk 1]
```diff
@@ -173,13 +173,13 @@ func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.Raw
 func (AppModule) ConsensusVersion() uint64 { return 5 }
 
 // BeginBlock returns the begin blocker for the ibc module.
-func (am AppModule) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
+func (am AppModule) BeginBlock(ctx sdk.Context) {
```
Comment (DimitrisJim) [inline modules/core/module.go:176]: ping, intentional inclusion in this PR?

Comment (faddat) [reply modules/core/module.go:176]: uninetntional, thank you

**Comment by faddat** [comment]
was definitely supposed to be in the 50 branch :)

**Review by damiannolan — APPROVED** [review]
ACK pending CI, lgtm! 🙏

**Review by DimitrisJim — APPROVED** [review]
nice, thanks!

**Review by chatton — APPROVED** [review]
LGTM

**Comment by colin-axner** [comment]
woah, awesome this could be pulled out, thanks @faddat! Also thanks @damiannolan @chatton @DimitrisJim for the ultra fast reviews! :zap:

**Pull request merged**